File: 백준/Silver/1026. 보물/보물.py
import sys

N = int(sys.stdin.readline())

A = list(map(int, sys.stdin.readline().split()))
B = list(map(int, sys.stdin.readline().split()))

A.sort()

sum = 0


for i in range(len(A)):
    temp = 0
    idx = 0
    for j in range(len(B)):
        if temp < B[j]:
            temp = B[j]
            idx = j
    sum += temp * A[i]
    B[idx] = -1

print(sum)

File: 백준/Silver/1026. 보물/보물.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.CopyOnWriteArrayList;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(bf.readLine());
        List<Integer> A = new ArrayList<>();

        String[] sa = bf.readLine().split(" ");
        for(int i=0; i<N; i++){
            A.add(Integer.parseInt(sa[i]));
        }
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        sa = bf.readLine().split(" ");
        for(int i=0; i<N; i++){
            pq.add(Integer.parseInt(sa[i]));
        }

        int answer = 0;
        Collections.sort(A, Collections.reverseOrder());
        for(int i=0; i<N; i++){
            answer += A.get(i) * pq.poll();
        }

        System.out.println(answer);
    }
}
